core-feature-pack/common/src/main/resources/content/bin/systemd/generate_systemd_unit.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/sh
+
+# This script is just a helper to generate the Systemd Unit Files for WildFly
+# assuming that the server home will be the one that hold this script file.
+# It also allows to specify the user and group that will run the server.
+
+SCRIPT_NAME=$(basename "$0")
+SCRIPT_DIR="$(dirname "$(realpath "$0")")"
+JBOSS_HOME="$(realpath "$SCRIPT_DIR/../..")"
+JBOSS_SYSTEMD_MODE="${1}"
+
+if [ -z "$JBOSS_SYSTEMD_MODE" ]; then
+  echo "INFO: Usage: "
+  echo "${SCRIPT_NAME} standalone|domain [user name] [group name]"
+  exit 0;
+fi
+
+SYSTEMD_FILE="$SCRIPT_DIR/wildfly-${JBOSS_SYSTEMD_MODE}.service"
+JBOSS_SYSTEMD_USER="${2:-wildfly}"
+JBOSS_SYSTEMD_GROUP="${3:-wildfly}"
+
+echo "INFO: Systemd Unit File to generate: ${SYSTEMD_FILE}"
+echo "INFO: Using JBOSS_HOME: $JBOSS_HOME"
+echo "INFO: User: ${JBOSS_SYSTEMD_USER}"
+echo "INFO: Group: ${JBOSS_SYSTEMD_GROUP}"
+
+while true; do
```

Comment:
Interactive shell is a bad idea. I would suggest using opts so that the user can pass a parameters if he wants to generate the systemd file.
We should also aimed at making this script idempotent. (I'll do another review in this regard, after this one)

Reply:
> Interactive shell is a bad idea.

It is not expected this file be invoked by a third-party tool, this is just a utility that is expected to be used by a user to make his life easier when he needs to replace the JBOSS_HOME, user:group in the systemd unit files. Asking interactively for confirmation here allows the user to review what he is going to configure, so I think it is useful here.

Would that convince you to leave it as it is?

Reply:
No :) I really think it's a bad idea. 

Bear in mind that this will be used by Ops (or DevOps), they won't expect an interactive shell and will be quite annoyed with it. Also, they are most likely going to automate around this (I've done so in the past, at customers). They may want to automate the deployment of Wildfly based and this and have script (or a tool like Ansible) automatically generate the file on the target system.... 

And right now, it's impossible to because the script is interactive by default. So if you really want to keep it, at least you should provide a mecanism around it.

IMHO, the best approach, that would match what people do in this kind of script, would be to test if the var is defined and act according to it:

```
if  [ -z ${SYSTEMD_FILE} ]; then
   echo "No systemd file provided, not generating one."
   exit 0
fi
```

(last point if somehow you really, really want to keep this interaction thing, I would move into a function, because it's big chunk of code)

Reply:
@rpelisse  Ok, you bring more experience to this from Ansible automation, so I'll remove the interactive question or add a confirm flag for skipping it

Reply:
@rpelisse I've added a new argument c,--confirm to avoid the interactive mode.